### ISSUE TITLE
nix-installer (Minimal Flake) gives error: at line # setup p10k.zsh for root in ./hosts/common/users/[user]/default.nix

### ISSUE THREAD

**Issue by DjCoke** [EmergentMind/nix-config#16]
### Problem Description

This bit of code causes an issue when `configVars.isMinimal` is set to `true`. The expectation is that the entire `home-manager` block is skipped, but the following error occurs: **`The option 'home-manager' does not exist`**.

Here is the problematic code:

```nix
# Setup p10k.zsh for root
home-manager.users.root = lib.optionalAttrs (!configVars.isMinimal) {
  home.stateVersion = "23.05"; # Avoid error
  programs.zsh = {
    enable = true;
    plugins = [
      {
        name = "powerlevel10k-config";
        src = configLib.relativeToRoot "home/${configVars.username}/common/core/zsh/p10k";
        file = "p10k.zsh";
      }
    ];
  };
};
```
So I tried all sorts of solutions, commenting it out stops the error, and while just for building the minimal flake is fine for the iso, it is not that I build it every day anyway, but I think here is an issue, or maybe I do something wrong and unexpected

I tried this:
```nix
# Setup p10k.zsh for root
home-manager = if configVars.isMinimal then { } else {
  home.stateVersion = "23.11"; # Avoid error (Changed by DjCoke: typo?)
  programs.zsh = {
    enable = true;
    plugins = [
      {
        name = "powerlevel10k-config";
        src = configLib.relativeToRoot "home/${configVars.username}/common/core/zsh/p10k";
        file = "p10k.zsh";
      }
    ];
  };
};
```

But this gives me the same error. I think just calling home-manager is sufficient to give this error I suppose?

So maybe the solution would be then like `fullUserConfig` to make a `fullRootConfig` in a let in statement?

**Comment by DjCoke** [comment]
The fix for me was to add this code to the `fullUserConfig` and that worked for me... But maybe it is maybe not the way you would want it...

**Comment by EmergentMind** [comment]
Thanks for pointing this out. I'm working on a substantial refactor that replaces configVars (which is sort of global to the entire config) with a hostSpec module that will allow per-host variables to be set. It should address this and a few other related bugs.

Will leave this open until then in case people are looking for a work around of sorts, which you noted.

**Comment by EmergentMind** [comment]
This issue should be resolved as of 31ab34e, however it hasn't been extensively tested due to #17 
Will track latent nixos-installer issues there.
